### PR TITLE
Implement #209 -- Validate route id reference in trips

### PR DESCRIPTION
**Pull request by lionel-nj**
**Summary:**

This PR provides support to check cross reference between files `trips.txt` and `routes.txt`.

**Expected behavior:** 

Records from `trips.txt` referring to non-existing records of `routes.txt` should generate a notice `NonExistingRouteIdNotice`.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "Fix #<issue_number> - <short description of fix and changes>" (for example - "Fix #1111 - Check for null value before using field")
- [x] Linked all relevant issues
- ~[ ] Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

File: RULES.md [hunk 1]
```diff
@@ -34,6 +34,7 @@ Rules are declared in the [`Notice` module](https://github.com/MobilityData/gtfs
 | [E030](#E030) | Inconsistent field `agency_timezone` | 
 | [E031](#E031) | Invalid `agency_id` | 
 | [E032](#E032) | `calendar.txt` `end_date` is before `start_date` |
+| [E032](#E033) | Reference to non-existing `route_id` in `trips.txt` |
```
Comment (barbeau) [inline RULES.md:37]: I would suggest we adopt a broader definition of this rule, like "`route_id` missing from routes.txt" or "Invalid route_id" or "route_id does not exist in routes.txt" that would cover all instances of a route_id used in a different file that doesn't exist in routes.txt.

route_id is used in many several files, and given the number of rules that will reference different IDs across different files, we're doing to end up with a lot of rules just of bad ID references. I think this unnecessarily clutters the rule list and makes it harder to find other semantic rules. Plus, we can trace which file the error is coming from from the message itself, right?

Also, related to the terminology of the rule name, we ended up using the following convention in the RT validator:
"GTFS-rt trip_id does not exist in GTFS data"

After a lot of discussion this seemed easier to understand that using words like "reference" or "non-existing" (which we also used at some point).

Comment (lionel-nj) [reply RULES.md:37]: True, and yes in the notice description we can track back the problematic file.

Comment (fabrice-v) [reply RULES.md:37]: I agree with the stated points, I suppose they are not implemented yet and will pause my review of this PR

File: RULES.md [hunk 2]
```diff
@@ -136,6 +137,12 @@ In `calendar.txt`, the `end_date` of a service record must not be earlier than t
 #### References:
 * [calendar.txt specification](https://gtfs.org/reference/static/#calendartxt)
 
+<a name="E033"/>
+
+### E033 - Invalid `route_id`
```
Comment (fabrice-v) [inline RULES.md:142]:
```suggestion
### E033 - `route_id` not found
```

File: usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripRouteIdTest.java [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+        underTest.execute();
+
+        verify(mockLogger, times(1)).info(ArgumentMatchers.eq(
+                "Validating rule E033 - Invalid `route_id`" + System.lineSeparator()));
+
+        verify(mockDataRepo, times(1)).getRouteAll();
+        verify(mockDataRepo, times(1)).getTripAll();
+
+        // suppress warning regarding ignored result of method since it is not necessary here.
```
Comment (fabrice-v) [inline usecase/src/test/java/org/mobilitydata/gtfsvalidator/usecase/ValidateTripRouteIdTest.java:62]: I think having just one explanatory comment would be enough. I'd suggest keeping that one and getting rid of the others

**Pull request merged**